packages/shared/src/components/modals/common/ModalClose.tsx

```diff
@@ -44,7 +45,9 @@ function ModalCloseComponent(
   }: ModalCloseProps,
   ref: Ref<HTMLButtonElement>,
 ): ReactElement {
-  if (!onClick) {
+  const { isDrawer } = useModalContext();
```

Comment:
Modal close button is not necessary in Drawers since it has its own close button.